include/zephyr/drivers/led.h

```diff
@@ -141,13 +141,12 @@ __syscall int led_blink(const struct device *dev, uint32_t led,
 static inline int z_impl_led_blink(const struct device *dev, uint32_t led,
 				   uint32_t delay_on, uint32_t delay_off)
 {
-	const struct led_driver_api *api =
-		(const struct led_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(led, dev));
```

Comment:
Offline discussion resulted in leaving out the asserts. The reason being that encapsulates APIs (eg I3c has i2c) can be used downstream.